### DISCUSSION TITLE
Is there a way to get SC to use multiple notes as context?

### DISCUSSION THREAD

**Discussion by avataraustin** [brianpetro/obsidian-smart-connections#666] [Q&A]
I keep noticing when doing queries that it always seems to show only 1 note used as context when I get an answer back.  In most of these cases I am certain that I had other notes with as good, if not better relevant information. It seems like it would be really helpful if the context had included some of these multiple sections.

**Comment by hittjwX** [comment]
I get good results mentioning 3 to 5 notes I want to consider in-scope for a prompt. I mention them with a wiki link. However, it doesn't use all of them or picks others, but the prompt returns good results.

**Reply by avataraustin** [reply]
So you need to link to the [[note(s) to use]] in your Smart Chat query for best results if I understand correctly? I was hoping  there was a way to have SC pick the top notes based on similarity score rather than apparently only choosing one.

**Reply by nomadphase** [reply]
you could use the Graph Analysis plugin as a first step in choosing notes based on similarity

**Reply by avataraustin** [reply]
Thanks, that is a really cool plugin I was not aware of.

**Reply by hittjwX** [reply]
When writing your prompt, the wiki link helps scope context. It seems nice to pick the top notes based on a similarity score, but that may skew a response unless your prompt consolidates those notes. 

If you have a note representing your audience, another of the services you offer, and a final description of the problems, then ask the AI to write an email about a scenario in which your service is ideal. There is no way for AI to build that context. These files aren't similar.

Having AI choose context from similarity scores (Jaccard or Overlap) will inbreed your response. Of course, if you have five files on amphibians and are writing an outline for an article on frogs, a similarity score may help provide the proper scope. 

I'd rather say, "Start with notes [[X]], [[Y]], and [[Z]], plus any contextually similar..." than omit the topical context relevant to my ask. You also bypass the need for excellent linking and front matter structure.
